flytekit/core/container_task.py

```diff
@@ -93,8 +99,148 @@ def __init__(
+        from flytekit.core.type_engine import TypeEngine, TypeTransformerFailedError
+        from flytekit.types.directory import FlyteDirectory
+        from flytekit.types.file import FlyteFile
+
+        try:
+            kwargs = translate_inputs_to_literals(
+                ctx,
+                incoming_values=kwargs,
+                flyte_interface_types=self.interface.inputs,
+                native_types=self.get_input_types(),  # type: ignore
+            )
+        except TypeTransformerFailedError as exc:
+            msg = f"Failed to convert inputs of task '{self.name}':\n  {exc}"
+            logger.error(msg)
+            raise TypeError(msg) from exc
+
+        input_literal_map = _literal_models.LiteralMap(literals=kwargs)
+        try:
+            native_inputs = self._literal_map_to_python_input(input_literal_map, ctx)
+        except Exception as exc:
+            msg = f"Failed to convert inputs of task '{self.name}':\n  {exc}"
+            logger.error(msg)
+            raise type(exc)(msg) from exc
+
+        # Normalize the input and output directories
+        if self._input_data_dir:
+            self._input_data_dir = os.path.normpath(self._input_data_dir)
+        if self._output_data_dir:
+            self._output_data_dir = os.path.normpath(self._output_data_dir)
+
+        # Create a temporary directory to store the output data
+        output_directory = ctx.file_access.get_random_local_directory()
+        volume_bindings = {
+            output_directory: {
+                "bind": self._output_data_dir,
+                "mode": "rw",
+            },
+        }
+
+        # Build the command string
+        commands = ""
+        cmd_and_args = []
+        if self._cmd:
+            cmd_and_args += self._cmd
+        if self._args:
+            cmd_and_args += self._args
+
+        for cmd in cmd_and_args:
+            k = self._get_key_from_cmd(cmd)
+            if k:
+                if type(native_inputs[k]) in [FlyteFile, FlyteDirectory]:
+                    local_flyte_file_or_dir_path = str(native_inputs[k])
+                    remote_flyte_file_or_dir_path = os.path.join(self._input_data_dir, k.replace(".", "/"))  # type: ignore
+                    volume_bindings[local_flyte_file_or_dir_path] = {
+                        "bind": remote_flyte_file_or_dir_path,
+                        "mode": "rw",
+                    }
+                    commands += remote_flyte_file_or_dir_path + " "
+                else:
+                    commands += str(native_inputs[k]).split()[0] + " "
+            else:
+                commands += cmd + " "
+
+        client = docker.from_env()
+        # If we can't find the image, pull it
+        if client.images.list(filters={"reference": self._image}) == []:
+            logger.info(f"Pulling image:{self._image} for container task:{self.name}")
+            client.images.pull(self._image)
+
+        container = client.containers.run(
```

Comment:
we should find a way to configure a timeout as well. This could be done in a separate PR, but can you leave a TODO here?